### PR TITLE
Cleaned up home page buttons

### LINKED ISSUE #420: Action buttons for each plio on the home page look ugly

### PR DESCRIPTION
**Pull request by dalmia**
Fixes #420 

## Test Plan

- [x] Test Responsiveness
   - [ ] Laptop (1200px)
   - [ ] Tablet (760px)
   - [ ] Phone (320px)
- [x] Cross-Browser Testing
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Safari
- [x] Local Language Support
- [ ] Wrote tests
- [x] Tested locally
- [ ] Tested on staging
- [ ] Tested on an actual physical phone
- [ ] Tested on production
- [x] Comments have been added appropriately
- [ ] Check for bundle size [here](https://bundlephobia.com/) if adding a package
- [x] Added relevant details like Labels/Projects/Milestones etc.

### CODE REVIEW

File: src/components/App/OptionDropdown.vue [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+      // sets the given index as the selected option index
+      if (this.isOptionDisabled(index)) return
+      this.toggleDropdownDisplay();
+      this.$emit('select', index, this.options[index].value)
+    },
+    isOptionsOverflowBottom() {
+        // checks whether the bottom of the options overflows the screen
+        const optionsContainer = document.querySelector('#options')
+        const rectangle = optionsContainer.getBoundingClientRect()
+        return rectangle.bottom > window.innerHeight
```
Comment (deepansh96) [inline src/components/App/OptionDropdown.vue:166]:
```suggestion
        return rectangle.bottom >= window.innerHeight
```
We don't want the option box to stick to the lower bottom as well

File: src/components/Collections/ListItems/PlioListItem.vue [hunk 2]
```diff
@@ -1,17 +1,41 @@
 <template>
   <PlioListItemSkeleton v-if="pending" class="w-full" />
-  <div v-else class="rounded-sm p-2 w-auto">
-    <div class="grid grid-flow-row auto-rows-min gap-2">
+  <div v-else class="flex relative rounded-sm p-2 w-auto">
```
Comment (deepansh96) [inline src/components/Collections/ListItems/PlioListItem.vue:3]: Why this `relative`? There is already a `relative` below? Two are needed?

File: src/components/Collections/ListItems/PlioListItem.vue [hunk 3]
```diff
@@ -135,36 +81,64 @@ export default {
...
-    // add listener for screen size being changed
-    window.addEventListener("resize", this.checkScreenOrientation);
+    // add listener for scrolling
+    window.addEventListener('scroll', this.handleScroll);
   },
-
   unmounted() {
     // remove listeners
-    window.removeEventListener("resize", this.checkScreenOrientation);
+    window.removeEventListener('scroll', this.handleScroll);
```
Comment (deepansh96) [inline src/components/Collections/ListItems/PlioListItem.vue:96]: Linter not working? Usually all single quotes are changed to double

File: src/components/Collections/ListItems/PlioListItem.vue [hunk 4]
```diff
@@ -135,36 +81,64 @@ export default {
...
+          disabled: !this.isPublished
+        })
+      }
+      let moreOptions = [
+        {
+          value: "duplicate",
+          label: this.$t("home.table.plio_list_item.buttons.duplicate"),
+          icon: "copy.svg",
+        },
+      ]
```
Comment (deepansh96) [inline src/components/Collections/ListItems/PlioListItem.vue:139]: Why another variable? Just add this duplicate waala object above where options are defined?

File: src/components/Collections/ListItems/PlioListItem.vue [hunk 5]
```diff
@@ -262,7 +170,9 @@ export default {
     },
     updatedAt() {
       // human readable date string
-      return new Date(this.plioDetails.updatedAt).toDateString();
+      // removed the first four characters of the output as they represent the day
+      // and lead to inconsistent length of the string
```
Comment (deepansh96) [inline src/components/Collections/ListItems/PlioListItem.vue:174]: best to not put "removed the first four..." thing in the comments. We know the old code so we can identify what has changed, but for someone reading in the future might not know that. Just add the date format as a comment

File: src/components/App/OptionDropdown.vue [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    },
+    convertRemToPixels(rem) {
+        // converts value in rem to pixels
+        return rem * parseFloat(getComputedStyle(document.documentElement).fontSize);
+    },
+    doesOptionHaveIcon(option) {
+      // whether the given option has an icon
+      return option != undefined && option.icon != null;
+    },
+    hideDropdown() {
```
Comment (deepansh96) [inline src/components/App/OptionDropdown.vue:151]: Some lines are not covered with tests. Will you write it or skip it for now?

**Review by deepansh96 — CHANGES_REQUESTED** [review]
Some comments, questions and one bug that I noticed 

- Bug on touch screens 
https://user-images.githubusercontent.com/20169829/131296202-d1495014-fca1-4f3d-a027-75d6126a76fc.mov

File: src/components/App/OptionDropdown.vue [hunk 7]
```diff
@@ -0,0 +1,199 @@
+<template>
```
[the remark this answers is not in the document]
Comment (dalmia) [reply src/components/App/OptionDropdown.vue:1]: meant to kept this as a generic dropdown only - isliye with `components/App`

File: src/components/App/OptionDropdown.vue [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+    isOptionDisabled(index) {
+        // whether the option at the given index is disabled
+        return this.options[index].disabled
+    },
+    setOptionBoxStyling() {
+        // sets the option box margin based on various parameters
+        if (!this.scrollY) {
+            this.optionBoxStyling = { 'margin-top': `${this.defaultOptionMarginRem}rem` }
+        } else {
+            this.optionBoxStyling = { 'margin-top': `-${this.scrollY - this.defaultOptionMargin}px`}
```
[the remark this answers is not in the document]
Comment (dalmia) [reply src/components/App/OptionDropdown.vue:140]: yes - scrollY will be in pixels.

File: src/components/App/OptionDropdown.vue [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    },
+    setOption(index) {
+      // sets the given index as the selected option index
+      if (this.isOptionDisabled(index)) return
+      this.toggleDropdownDisplay();
+      this.$emit('select', index, this.options[index].value)
+    },
+    isOptionsOverflowBottom() {
+        // checks whether the bottom of the options overflows the screen
+        const optionsContainer = document.querySelector('#options')
```
[the remark this answers is not in the document]
Comment (dalmia) [reply src/components/App/OptionDropdown.vue:164]: the same thing can happen with ref na?

File: src/components/App/OptionDropdown.vue (hunk 6 above)
Comment (dalmia) [reply src/components/App/OptionDropdown.vue:151]: need to simulate clickaway for this line which I am not sure how to do!

File: src/components/App/OptionDropdown.vue (hunk 7 above)
[the remark this answers is not in the document]
Comment (dalmia) [reply src/components/App/OptionDropdown.vue:1]: rename old to select dropdown

File: src/components/App/OptionDropdown.vue (hunk 9 above)
[the remark this answers is not in the document]
Comment (dalmia) [reply src/components/App/OptionDropdown.vue:164]: `dropdownOptions`

File: src/components/App/OptionDropdown.vue [hunk 10]
```diff
@@ -0,0 +1,199 @@
+<template>
+  <div>
+    <div class="flex justify-end">
```
[the remark this answers is not in the document]
Comment (dalmia) [reply src/components/App/OptionDropdown.vue]: removing this is messing up the positioning of the options container

**Pull request merged**